docs/design/maven.md

```diff
@@ -0,0 +1,141 @@
+time and bandwidth.
+
+### Maven artifacts
+
+[Maven artifacts][maven-artifacts] are defined as anything that resides in a maven repository. Maven has a special way of
+addressing artifacts in its repositories called coordinates, which are usually represented by a `GAV` tuple (stands for
+'Group, Artifact, Version').
+This is used to uniquely identify an artifact in a maven repository.
+
+### Maven purls
+
+Purls for `pkg:maven` ([docs][maven-purl]) fully describe the location of an artifact in a maven repository. That is because
+all parts of the purl (`namespace`, `name`, `version` and the `classifier` and `type` qualifiers) are used to construct
+the download url for the artifact. [Maven documentation][maven-repo-layout] describes how to turn these coordinates into
+an actual download url. Here's an example:
+
+**Purl:**
+
+```
+pkg:maven/ga.io.quarkus/quarkus-core@3.8.5.redhat-00004?type=jar&repository_url=https://maven.repository.redhat.com&checksums=sha1:e4ca5fadf89e62fb29d0d008046489b2305295bf
+```
+
+**Mapping to GAV:**
+
+```
+groupId: ga.io.quarkus
+artifactId: quarkus-core
+version: 3.8.5.redhat-00004
+classifier: null
+type: jar
+```
+
+Using `classifier` and `type` we can determine the `extension` is `jar` ([docs][maven-extension]).
+
+**Download url**:
+
+Template:
+
+```
+${groupId as directory}/${artifactId}/${baseVersion}/${artifactId}-${version}.${extension}
+```
+
+Resulting url:
+
+```
+https://maven.repository.redhat.com/ga/io/quarkus/quarkus-core/3.8.5.redhat-00004/quarkus-core-3.8.5.redhat-00004.jar
+```
+
+## Design
+
+In this section, I will try to cover individual parts of the design.
+
+### Source repository (cachi2 lockfile)
+
+Currently, the cachi2 lockfile only allows `pkg:generic` purl type. This design will introduce a new purl type `pkg:maven`
+to support pre-fetching individual maven artifacts. The lockfile currently supports specifying the file to be fetched either
+as a purl or as a pair of `download_url` and `checksums`. Maven artifacts should only be specified as a purl in order to
+be reported as such in the output SBOM. Here's an example of a lockfile that fetches a maven artifact:
+
+```yaml
+metadata:
+  version: 1.0.0
+artifacts:
+  - purl: pkg:maven/ga.io.quarkus/quarkus-core@3.8.5.redhat-00004?type=jar&repository_url=https://maven.repository.redhat.com&checksums=sha1:e4ca5fadf89e62fb29d0d008046489b2305295bf
+    target: quarkus.jar
+```
+
+### Integration with existing cachi2 features
+
+As mentioned above, the lockfile format is identical to the one used for describing generic artifacts, therefore it makes
+sense to extend the existing lockfile with support for maven purls. The only difference would be in the output SBOM where
+maven artifacts would be reported as `pkg:maven`, instead of `pkg:generic`.
+
+There is an open question of how the UX should be. I will try to list the available options with some pros and cons.
```

Comment:
In my opinion, the part that changes is:
- how you compute the download url (meaning, the list of accepted purl types _might_ be expanded later)
- what the output SBOM component is (depends on the above) 

So I'd argue that since the underlying mechanism does not change, it should stay as one package manager.